apps/site/assets/js/event-page-setup.js

```diff
@@ -29,9 +31,64 @@ function addInternetExplorerPolyfills() {
   stickyPolyfill();
 }
 
+function setupEventPopups() {
+  /**
+   * Why not use Bootstrap's popover? It's not equipped to handle having lots of
+   * interactive content, doesn't support a close button, and as implemented the
+   * UX for assistive tech users probably isn't ideal. For more, see "Making
+   * popovers work for keyboard and assistive technology users" in
+   * https://getbootstrap.com/docs/4.6/components/popovers/
+   *
+   * This implementation uses ally-dialog (https://a11y-dialog.netlify.app) as a
+   * base to create accessible dialog windows for various use cases.
+   */
+  const calendarView = document.querySelector(".m-events-hub--list-view");
+
+  // find all the event popup HTML elements
+  const eventPopups = calendarView.querySelectorAll(".m-event-overlay");
+  for (let i = 0; i < eventPopups.length; i++) {
+    const el = eventPopups[i];
+    const dialog = new A11yDialog(el);
+
+    // Hide popup when user clicks outside popup, or on a different event
+    document.addEventListener("click", e => hideDialogLogic(e, dialog));
+    document.addEventListener(
+      "turbolinks:before-render",
+      document.removeEventListener("click", e => hideDialogLogic(e, dialog)),
```

Comment:
This doesn't quite work, since the two functions aren't the same by identity. You need something like:
```
const listener = e => hideDialogLogic(e, dialog);
document.addEventListener("click", listener);
document.addEventListener("turbolinks:before-render", () => document.removeEventListener("click", listener));
```

Reply:
Oh my goodness, I remember this stuff!  Thanks for catching